File: mario.c
#include <stdio.h>
#include <cs50.h>

void build_pyramid(int width);

int main(void)
{
    // initialize height variable
    int height;

    // Ask user for height input and re-prompt if input is invalid
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Loop through positive numbers from 1 until input height
    for (int i = 1; i <= height; i++)
    {
        // Calculate spaces for row
        int spaces = height - i;

        // Print a space character for each of the spaces
        for (int j = 0; j < spaces; j++)
        {
            printf(" ");
        }

        // Builds row for left pyramid
        build_pyramid(i);

        // Print the 'gap'
        printf("  ");

        // Builds row for right pyramid
        build_pyramid(i);

        // Finish the row with a new line
        printf("\n");
    }
}

void build_pyramid(int width)
{
    // Print a hash for each of the outer loop's count
    for (int i = 0; i < width; i++)
    {
        printf("#");
    }
}

File: mario.py
from cs50 import get_int


def main():

    # Initialize height to zero
    height = 0

    # Whilst height is less than or equal to zero or greater than eight, ask user for integer
    while height <= 0 or height > 8:
        height = get_int("Height: ")

    # Loop numbers from 1 to height (height + 1 as range only creates a sequences up until but not including the last number)
    for i in range(1, height + 1):
        spaces = height - i
        blocks = height - spaces

        print_spaces(spaces)
        build_pyramid(blocks, "")
        print_spaces(2)
        build_pyramid(blocks, "\n")


# Print blocks for each half of pyramid
def build_pyramid(width, end):
    print("#" * width, end=end)


# Print n number of spaces
def print_spaces(n):
    print(" " * n, end="")


main()